### ISSUE TITLE
feature request: "dump" method in [text]

### ISSUE THREAD

**Issue by jmmmp** [pure-data/pure-data#867]
With a dump method, it's possible to get the complete contents of a text, without building some specific circuit for it.

**Comment by Spacechild1** [comment]
What exactly do you mean with "complete contents"? In which format would you except to get content?

`[text tolist]` already outputs the whole content as a single list (with semis and commas escaped accordingly).

**Comment by porres** [comment]
I guess all lines sequentially?

**Comment by Spacechild1** [comment]
What about seperators? And what's the use case?

BTW, there's already `[text tolist]`, but that's probably not what @jmmmp wants.

**Comment by Spacechild1** [comment]
BTW, that's probably the easiest way to get the whole text as a single list:

![grafik](https://user-images.githubusercontent.com/16126632/160633605-74312d80-8044-4df7-8b8a-5a13624ac3f4.png)

A "text dump" method could do this a bit more efficiently (i.e. omit the final list copy), but I'm not sure it's a common enough operation to optimize for.

**Comment by reduzent** [comment]
Isn't that what [text tolist] does? It does it in a way so that the whole text buffer can be restored with [text fromlist].

If [text dump] is meant to output all lines sequentially, then there is already [text sequence]. I don't see what  [text dump] could bring to the table that is new.

**Comment by Spacechild1** [comment]
> Isn't that what [text tolist] does?

Not really. `[text tolist]` converts the binbuf to plain text, i.e. semi and comma atoms are converted to symbols and special characters inside symbols are escaped.

**Comment by jmmmp** [comment]
Hi,

basically it would be the same function as dump in cyclone/coll. Which is the contrary to [text tolist]: to get all lines in a text object with one click. I need this often to parse data from textfiles, and usually have to build a [until]-contraption (which either reads the number of lines before banging, or stops it with a right-inlet bang). But it seems to me that dumping all contents at once would be an important use-case to add another method - as it was in coll.

Joao

**Comment by Spacechild1** [comment]
@jmmmp 

> Which is the contrary to [text tolist]: to get all lines in a text object with one click.

The opposite of `[text tolist]` is `[text fromlist]`.

> basically it would be the same function as dump in cyclone/coll. 

The `[dump(` method for `[cyclone/coll]` outputs all lines *sequentially*. This is the same as sending `[line 0, bang(` to `[text sequence]`. So what are you missing exactly?

**Comment by porres** [comment]
we could make that more explicit as a functionality in the documentation

**Comment by jmmmp** [comment]
basically it would be the same function as dump in cyclone/coll.

The [dump( method for [cyclone/coll] outputs all lines *sequentially*. This
is the same as sending [line 0, bang( to [text sequence]. So what are you
missing exactly?

I was never missing anything, as I always managed to find a way to get the
output I wanted. Being what I could think a general use case, it might be
better to make it more available.

I don't use [text sequence], as that concept isn't that interesting to me.
But if something is there and now is clear, I can have a quick look to see
if it works.

**Comment by Spacechild1** [comment]
> I don't use [text sequence], as that concept isn't that interesting to me.

You should really check the documentation of an object before making a feature request...

> But if something is there and now is clear, I can have a quick look to see if it works.

Yes please :-) I've already posted an example: https://github.com/pure-data/pure-data/issues/867#issuecomment-1081941152

**Comment by porres** [comment]
Trying to think of a new 'dump' message...

maybe create a 'dump' message for [text sequence] that would be the same as "line 0, bang" but would also ignore waits (not stopping when reaching them).

**Comment by Spacechild1** [comment]
> but would also ignore waits (not stopping when reaching them).

But that's what `[line 0, bang(` already does?

**Comment by porres** [comment]
> But that's what [line 0, bang( already does?

nope, if you have waits, it'll stop at them.

**Comment by porres** [comment]
yeah, the whole thing says "This is a zero length list. It output all lines starting from current one to next waiting point (but note there aren't any waits in this example so all lines are output)"

and yeah, you need flags to specify waits.

so, my idea of a dump message would be to ignore waits and output them all sequentially anyway

**Comment by porres** [comment]
there's a typo there, should be "outputs"

**Comment by Spacechild1** [comment]
> yeah, the whole thing says "This is a zero length list. It output all lines starting from current one to next waiting point (but note there aren't any waits in this example so all lines are output)"

Ah, I didn't look at the most recent Pd version :-) So this is already covered.

> so, my idea of a dump message would be to ignore waits and output them all sequentially anyway

But we already have `[line 0, bang( -> [text sequence]`... Why do we need a "dump" method?

Maybe the `[line 0, bang(` "trick" could be mentioned explicitly somewhere in the `[text]` documentation?

**Comment by Spacechild1** [comment]
> In my idea, it would output everything EVEN if we have waits, cause it'll stop now if we have waits...

??? You you don't want to use waits, you wouldn't write `[text sequence -w]` in the first place....

`[line 0, bang( -> [text sequence]` does *exactly* what a "dump" method would do...

**Comment by porres** [comment]
But I have to say that a "dump" message to plain [text] ("text define") wouldn't be bad!

**Comment by Spacechild1** [comment]
> But I have to say that a "dump" message to plain [text] ("text define") wouldn't be bad!

`[text define]` and `[array define]` are just there to *define* an object. Then you have several "method" to interact with the object. So I'm pretty sure that `[text define]` should *not* get a "dump" method.

I would agree that `[line 0, bang( -> [text sequence]` might not be the most obvious mechanism, but I think it's largely a matter of documentation.

**Comment by Spacechild1** [comment]
> I don't have a use case, but it would allow one to use both

??? You can have both `[text sequence]` and `[text sequence -w]` in the same patch, depending on what you want to do...

I think it's always a good idea to come up with a use case before proposing a feature ;-)
